x-pack/plugins/session_view/public/components/process_tree_alerts_filter/index.tsx

```diff
@@ -90,7 +91,13 @@ export const ProcessTreeAlertsFilter = ({
       onClick={onButtonClick}
     >
       <EuiText size="s">
-        {i18n.SELECTED_ALERT_CATEGORY_TEXT(selectedProcessEventAlertCategory)}
+        <FormattedMessage
+          id={`sessionView.alertDetailsFilterItem-${selectedProcessEventAlertCategory}`}
+          defaultMessage="View: {selectedProcessEventAlertCategory} alerts"
```

Comment:
Ok, so will this generate entries for each type with the full text (without template param)? or will the text to translate be. `View: {selectedProcessEventAlertCategory} alerts"` for each of the 3 id types?

I think you can run `node scripts/i18n_extract --output-dir ./` to check.

Reply:
I ran it locally, and notice that all template params stay in the compiled output (since they are meant for dynamic run time changes to the string). You could make this approach work, but the selectedProcessEventAlertCategories would all need to be localized individually as well. Otherwise this localization would be translated except for the dynamic portion.

Reply:
You'll most likely end up with 3 identical entries in the en.json file. e.g
"sessionView.alertDetailsFIlterItem-process": "View {processEventAlertCategory} alerts"
"sessionView.alertDetailsFIlterItem-file": "View {processEventAlertCategory} alerts"
"sessionView.alertDetailsFIlterItem-network": "View {processEventAlertCategory} alerts"

Reply:
that way the defaultMessage will be rendered as expected at extraction time.

Reply:
Hmm default message property is looking for string type. If i try  the template literal there, then I have to wrap template literal in an object which incompatible type for default message property.
<img width="734" alt="image" src="https://user-images.githubusercontent.com/17135495/200952119-e6079196-64ce-4a6b-b9fa-1c619923d300.png">

Reply:
<img width="921" alt="image" src="https://user-images.githubusercontent.com/17135495/200953139-a02fd482-039e-42c4-bb19-d67a281f7d78.png">

Reply:
yea, that makes sense. I tried with the translate function too and it prevents it.
![image](https://user-images.githubusercontent.com/16198204/200955481-4f6e2c32-67fa-4924-b81c-4ef8ed17ebba.png)

![image](https://user-images.githubusercontent.com/16198204/200954203-fd286bc1-9afb-4164-a3f5-7e9e8e14a512.png)

Same with the id:
![image](https://user-images.githubusercontent.com/16198204/200955238-3c6b6301-d8cd-44fe-ae0c-a1ef4c7639d5.png)

This convinces me the library doesn't like localizations to be built dynamically. Except for dynamic parts of a message that don't require translation. Like a count. or switching on plural vs singular form. Those template vars will be contained in the en.json after extraction.

e.g

en.json
"xpack.sessionView.viewAlertType": "View {type} alert"

in french would be:
fr.json
"xpack.sessionView.viewAlertType": "Afficher l'alerte de {type}"

Type is created at runtime, and so french users would see "Afficher l'alerte de (file|network|process)". You could translate those 3 values individually (3 more localizations), but then you might as well just go with 3 distinct localizations for each alert type. I think it's important the full context of the button text be present to the translator, or it could result in some interesting grammar :)

Reply:
@mitodrummer, following that line of thought, to not lose context, it makes sense to create separate sentences for each case:

```js
import { i18n } from "@kbn/i18n";

export const SELECTED_PROCESS = {
  all : i18n.translate("sessionView.alertDetailsFIlterItem-all", {
    defaultMessage: "View: All alerts",
  },
  process : i18n.translate("sessionView.alertDetailsFIlterItem-process", {
    defaultMessage: "View: Process alerts",
  },
  network : i18n.translate("sessionView.alertDetailsFIlterItem-network", {
    defaultMessage: "View: Network alerts",
  },
  file : i18n.translate("sessionView.alertDetailsFIlterItem-file", {
    defaultMessage: "View: File alerts",
  })
}
```


And then use it like :

```jsx
<EuiButtonEmpty
      data-test-subj="sessionView:sessionViewAlertDetailsEmptyFilterButton"
      size="s"
      iconType="arrowDown"
      iconSide="right"
      onClick={onButtonClick}
    >
      <EuiText size="s">
          {SELECTED_PROCESS[selectedProcessEventAlertCategory]}
     </EuiText>
</EuiButtonEmpty>
```

Reply:
We just need to be prepared for edge cases:

- i.e new or different alert types (other than network, process or file)

```js
import { i18n } from "@kbn/i18n";

export const SELECTED_PROCESS = {
  other : i18n.translate("sessionView.alertDetailsFIlterItem-other", {
    defaultMessage: "View: Other alerts"
  })
}
```

```jsx
<EuiButtonEmpty
      data-test-subj="sessionView:sessionViewAlertDetailsEmptyFilterButton"
      size="s"
      iconType="arrowDown"
      iconSide="right"
      onClick={onButtonClick}
    >
      <EuiText size="s">
          {SELECTED_PROCESS[selectedProcessEventAlertCategory] ? SELECTED_PROCESS[selectedProcessEventAlertCategory] : SELECTED_PROCESS.other}
     </EuiText>
</EuiButtonEmpty>
```